test/e2e/app-dir/metadata-icons/metadata-icons.test.ts

```diff
@@ -0,0 +1,31 @@
+import { nextTestSetup } from 'e2e-utils'
+
+describe('app-dir - metadata-icons', () => {
+  const { next } = nextTestSetup({
+    files: __dirname,
+  })
+
+  it('should only have 1 favicon link in root page', async () => {
+    const $ = await next.render$('/')
+    expect($('link[href^="/favicon.ico"]').length).toBe(1)
+  })
+
+  it('should only have 1 favicon link in nested page', async () => {
+    const $ = await next.render$('/nested')
+    expect($('link[href^="/favicon.ico"]').length).toBe(1)
+  })
+
+  it('should render custom icons along with favicon in root page', async () => {
+    const $ = await next.render$('/')
+    expect($('link[rel="shortcut icon"]').attr('href')).toBe(
+      '/shortcut-icon.png'
+    )
+  })
+
+  it('should render custom icons along with favicon in nested page', async () => {
+    const $ = await next.render$('/nested')
+    expect($('link[rel="shortcut icon"]').attr('href')).toBe(
+      '/shortcut-icon-nested.png'
+    )
+  })
```

Comment:
should we warn then? It's not clear from the commits what exactly changed why it should just silently change.

Reply:
I don't think we need to warn anything as the favicon.ico should be presented everywhere. If you want to show different icons for different pages, you can use different metadata exports instead of having a favicon.ico convention for global use case.

Reply:
Can't review when it's not clear what changed. Having one commit for current behavior and one commit with fix+test change would help. Anything beyond that requires too much time for reviewers unfamiliar with the behavior.